lib/common/digest.c

```diff
@@ -240,3 +240,47 @@ crm_digest_verify(xmlNode *input, const char *expected)
     free(calculated);
     return passed;
 }
+
+char*
+versioned_param_summary_for_digest(xmlNode *versioned_params, const char *name)
+{
+    xmlNode *attrs = NULL;
+    char *summary = NULL;
+    gboolean found = FALSE;
+
+    if (!name) {
+        return NULL;
+    }
+
+    for (attrs = __xml_first_child(versioned_params); attrs != NULL; attrs = __xml_next_element(attrs)) {
+        xmlNode *rule = __xml_first_child(attrs);
+        const char *boolean_op = crm_element_value(rule, XML_RULE_ATTR_BOOLEAN_OP);  
+        xmlNode *expr = NULL;
+        xmlNode *attr = NULL;
+
+        if (boolean_op == NULL) {
+            boolean_op = "and";
+        }
+        summary = add_list_element(summary, boolean_op);
+
+        for (expr = __xml_first_child(rule); expr != NULL; expr = __xml_next_element(expr)) {
+            summary = add_list_element(summary, crm_element_value(expr, XML_EXPR_ATTR_OPERATION));
+            summary = add_list_element(summary, crm_element_value(expr, XML_EXPR_ATTR_VALUE));
```

Comment:
Well, I thought that value doesn't matter for calculating digest. I only use add_list_element() for convenient accumulation of all versioned parameter parts. Doesn't any change in value lead to different digest?